staging/kos/pkg/controllers/connectionagent/connection_agent.go

```diff
@@ -1052,15 +1054,26 @@ func (ca *ConnectionAgent) localAttachmentIsUpToDate(att *netv1a1.NetworkAttachm
 }
 
 func (ca *ConnectionAgent) updateLocalAttachmentStatus(att *netv1a1.NetworkAttachment, macAddr, ifcName string, statusErrs sliceOfString, pcer *netv1a1.ExecReport) error {
-	att2 := att.DeepCopy()
-	att2.Status.MACAddress = macAddr
-	att2.Status.IfcName = ifcName
-	att2.Status.HostIP = ca.hostIP.String()
-	att2.Status.Errors.Host = statusErrs
-	att2.Status.PostCreateExecReport = pcer
+	status := netv1a1.NetworkAttachmentStatus{
+		MACAddress:           macAddr,
+		IfcName:              ifcName,
+		HostIP:               ca.hostIP.String(),
+		Errors:               netv1a1.NetworkAttachmentErrors{Host: statusErrs},
+		PostCreateExecReport: pcer,
+	}
+	var pb bytes.Buffer
+	pb.WriteString(`{"status":`)
+	encoder := json.NewEncoder(&pb)
+	err := encoder.Encode(status)
+	pb.WriteString(`}`)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to Encode(%#+v): %#+v", status, err))
+	}
+	patchBytes := pb.Bytes()
+	klog.V(8).Infof("Patching %s/%s by %q", att.Namespace, att.Name, string(patchBytes))
 
 	tBeforeUpdate := time.Now()
-	updatedAtt, err := ca.netv1a1Ifc.NetworkAttachments(att2.Namespace).Update(att2)
+	updatedAtt, err := ca.netv1a1Ifc.NetworkAttachments(att.Namespace).Patch(att.Name, k8stypes.StrategicMergePatchType, patchBytes, "status")
```

Comment:
1.  Yes.  As you can see in the status strategy's PrepareForUpdate, it is only the Spec that an update cannot change.  BTW, I would recommend keeping the implementation's state in Status rather than ObjectMeta.Annotations.

2. The patch strategy annotations are needed for lists because they have various semantics.  The a strategic merge patch handles a map the same was a JSON merge patch does --- key by key.

Reply:
1) No objection to storing the new field in status rather than in an annotation.

2) Oh, right. Sorry, I was reasoning as if `ObjectMeta.Annotations` was a list rather than a map.